storage/ndb/rest-server/data-access-rondb/src/rdrs-dal.cpp

```diff
@@ -104,15 +124,22 @@ RS_Status pk_read(RS_Buffer *reqBuff, RS_Buffer *respBuff) {
     return status;
   }
 
-  PKROperation pkread(reqBuff, respBuff, ndb_object);
+  Uint32 orid = OP_RETRY_INITIAL_DELAY;
+  Int32 orc   = OP_RETRY_COUNT + 1;
+  do {
+    PKROperation pkread(reqBuff, respBuff, ndb_object);
+    status = pkread.PerformOperation();
+    --orc;
+    if (orc > 0 && CanRetryOperation(status)) {
```

Comment:
Still, why do want to check whether you can retry an operation if it has potentially succeeded?

Just call `CanRetryOperation()` if `httpStatus != 200`?

Reply:
fixed